solr_conf/conf/managed-schema

```diff
@@ -258,6 +258,18 @@
       </analyzer>
     </fieldType>
 
+    <!-- Simple text field with no stemming, used to find exact matches. -->
+    <fieldType name="text_nostem" class="solr.TextField">
+      <analyzer>
+        <tokenizer class="solr.WhitespaceTokenizerFactory"/>
+        <filter class="solr.LowerCaseFilterFactory"/>
+        <filter class="solr.ICUFoldingFilterFactory"/>
+        <!-- ensure hyphenated words are joined together, even if they were separated by whitespace -->
+        <filter class="solr.HyphenatedWordsFilterFactory"/>
+        <!-- remove all punctuation from the end of tokens -->
+        <filter class="solr.PatternReplaceFilterFactory" pattern="[,\.!?:;]$" replacement=""/>
```

Comment:
Is there a punctuation pattern match (similar to `\d` for numeric) we can use here? There are probably other unicode punctuation that you might need to include.

Reply:
Looks like we might be able to do `\p{Punct}` or `\p{IsPunctuation}`. There's [some interesting complexity](https://stackoverflow.com/a/49289766) with those but the two in combination should cover everything we need. I had avoided doing something like this due to the inclusion of the hyphen, but now I realize that should actually be completely resolved by the previous filter and hyphens that are truly at the end of the token _should_ be discarded.